File: cpp/Sort/selectSort.cpp
#include "Sort.h"

int Sort::selectSort(vector<int> &nums, bool asc) {
	for (int i = 0; i < nums.size()-1; i++) {
		for (int j = i + 1; j < nums.size(); j++) {
			if (nums[j] < nums[i]) Sort::swap(&nums[i], &nums[j]);
		}
	}
	if (!asc) Sort::reverse(nums);
	return 0;
}

File: js/Sort/selectSort.js
function swap(arr, i, j) {
    var tmp = arr[i];
    arr[i] = arr[j];
    arr[j] = tmp;
    return arr;
}

function reverse(nums) {
    for (var i=0; i<parseInt(nums.length / 2); i++) nums = swap(nums, i, nums.length-1-i);
    return nums;
}

function selectSort(nums, asc) {
    for (var i=0; i<nums.length; i++) {
        for (var j=i+1; j<nums.length; j++) {
            if (nums[i] > nums[j]) nums = swap(nums, i, j);
        }
    }
    if (!asc) nums = reverse(nums);
    return nums;
}

var res = selectSort([2,4,1,2], false);
console.log(res);